src/ucp/wireup/address.c

```diff
@@ -751,40 +1100,46 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
     int attr_len;
     void *ptr;
     int enable_amo;
+    uint8_t addr_flags;
 
     ptr               = buffer;
     addr_index        = 0;
+    addr_flags        = 0;
     address_header_p  = ptr;
-    *address_header_p = UCP_ADDRESS_VERSION_CURRENT;
-    ptr               = UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
+    *address_header_p = addr_version;
+    ptr               = (addr_version == UCP_ADDRESS_VERSION_V1) ?
+                        UCS_PTR_TYPE_OFFSET(ptr, uint8_t) :
+                        UCS_PTR_TYPE_OFFSET(ptr, uint16_t);
 
     if (pack_flags & UCP_ADDRESS_PACK_FLAG_AM_ONLY) {
-        *address_header_p |= UCP_ADDRESS_HEADER_FLAG_AM_ONLY;
+        addr_flags |= UCP_ADDRESS_HEADER_FLAG_AM_ONLY;
     }
 
     if (pack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
-        *(uint64_t*)ptr    = worker->uuid;
-        ptr                = UCS_PTR_TYPE_OFFSET(ptr, uint64_t);
-        *address_header_p |= UCP_ADDRESS_HEADER_FLAG_WORKER_UUID;
+        *(uint64_t*)ptr = worker->uuid;
+        ptr             = UCS_PTR_TYPE_OFFSET(ptr, uint64_t);
+        addr_flags     |= UCP_ADDRESS_HEADER_FLAG_WORKER_UUID;
     }
 
     if (pack_flags & UCP_ADDRESS_PACK_FLAG_CLIENT_ID) {
-        *(uint64_t*)ptr    = worker->client_id;
-        ptr                = UCS_PTR_TYPE_OFFSET(ptr, uint64_t);
-        *address_header_p |= UCP_ADDRESS_HEADER_FLAG_CLIENT_ID;
+        *(uint64_t*)ptr = worker->client_id;
+        ptr             = UCS_PTR_TYPE_OFFSET(ptr, uint64_t);
+        addr_flags     |= UCP_ADDRESS_HEADER_FLAG_CLIENT_ID;
     }
 
     if (worker->context->config.ext.address_debug_info) {
         /* Add debug information to the packed address, and set the corresponding
          * flag in address header.
          */
-        *address_header_p |= UCP_ADDRESS_HEADER_FLAG_DEBUG_INFO;
+        addr_flags |= UCP_ADDRESS_HEADER_FLAG_DEBUG_INFO;
 
         if (pack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_NAME) {
             ptr            = ucp_address_pack_worker_address_name(worker, ptr);
         }
     }
 
+    ucp_address_pack_header_flags(address_header_p, addr_version, addr_flags);
```

Comment:
maybe have one function ucp_address_pack_header(ptr, version, flags)
and remove line 1109

Reply:
will try with newly introduced header_v2

Reply:
struct for header v2 does not look better (just two uint8_t)
here, I have to work with flags after deciding what size is the header, therefore need to pack flags later